File: maven-root/mtsa/src/main/java/RobotProtocol/src/main/java/robotProtocol/messages/CommunicationStartAcknowledgeMessage.cpp
/*
 * CommunicationStartAcknowledgeMessage.cpp
 *
 *  Created on: Feb 3, 2013
 *      Author: mariano
 */

#include "CommunicationStartAcknowledgeMessage.h"

namespace RobotProtocol {

CommunicationStartAcknowledgeMessage CommunicationStartAcknowledgeMessage::staticMessage = CommunicationStartAcknowledgeMessage(0,0);

Message *CommunicationStartAcknowledgeMessage::deserialize(char *serializedMessage){
	CommunicationStartAcknowledgeMessage::staticMessage.initialize(serializedMessage);
	return &(CommunicationStartAcknowledgeMessage::staticMessage);
}

CommunicationStartAcknowledgeMessage::CommunicationStartAcknowledgeMessage(int to, int from)
:Message(MESSAGE_ID, 1, to, from){}

CommunicationStartAcknowledgeMessage::CommunicationStartAcknowledgeMessage(char *serializedMessage){
	initialize(serializedMessage);
}

CommunicationStartAcknowledgeMessage::~CommunicationStartAcknowledgeMessage() {
}

int CommunicationStartAcknowledgeMessage::getMessageLength(){
	return Message::getMessageLength();
}

void CommunicationStartAcknowledgeMessage::readData(char **serializedMessageIndex){}
void CommunicationStartAcknowledgeMessage::writeData(char **serializedMessageIndex){}

} /* namespace RobotProtocol */

File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/CommunicationStartAcknowledgeMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class CommunicationStartAcknowledgeMessage extends Message {	
	public static int MESSAGE_ID				= 2;
	
	public CommunicationStartAcknowledgeMessage(int to, int from){
		super(MESSAGE_ID, 1, to, from);
	}
	
	public CommunicationStartAcknowledgeMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength();
	}
	
	
	@Override
	protected void readData(String inputString){
	}

	@Override
	protected String writeData(){
		return "";
	}

	@Override
	protected boolean hasReply() {
		return false;
	}
	
}
